tokio-util/src/compat.rs

```diff
@@ -232,7 +232,9 @@ impl<T: tokio::io::AsyncSeek> futures_io::AsyncSeek for Compat<T> {
             *self.as_mut().project().seek_pos = Some(pos);
         }
         let res = ready!(self.as_mut().project().inner.poll_complete(cx));
-        *self.as_mut().project().seek_pos = None;
+        if res.is_ok() {
+            *self.as_mut().project().seek_pos = None;
+        }
```

Comment:
Why is this being changed? Is there a use-case where this behaves incorrectly? Can you add that use-case as a test?